src/python/pants/base/deprecated.py

```diff
@@ -53,34 +57,34 @@ def get_deprecated_tense(removal_version, future_tense='will be', past_tense='wa
 
 
 @memoized_method
-def validate_removal_semver(removal_version):
-  """Validates that removal_version is a valid semver.
+def validate_deprecation_semver(version_string, version_description):
```

Comment:
This new parameter `version_description` confuses me a bit. It seems every call to this method fixes the value to `removal version`? Is there any reason you parametrized it?

If we stick with the parameter, a docstring would be helpful enumerating what you are expecting. Even better would be an Enum, but Py2 doesn't have that yet, so a docstring would be good for now.

Reply:
This was added (along with the method renamed) because the `deprecation_start_version` passes in a different value.

Reply:
I'll definitely add a docstring though, oops.

Reply:
Added a docstring!